### DISCUSSION TITLE
*** ERROR *** Cannot allocate a NumericVector

### DISCUSSION THREAD

**Discussion by ykvishal** [idaholab/moose#18386] [Q&A Meshing] [answered]
I was trying to run a large 3D grain growth simulation  ([3D_6000_gr.i](https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/grain_growth/3D_6000_gr.i))  and I am getting following error

> *** ERROR ***
Cannot allocate a NumericVector with 3224060050 degrees of freedom. The vector can only index up to 2147483647 entries.

>--------------------------------------------------------------------------
>MPI_ABORT was invoked on rank 60 in communicator MPI_COMM_WORLD
with errorcode 1.

>NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.

My mesh block is shown here
```
[Mesh]
  [gmg]
    type = DistributedRectilinearMeshGenerator
    dim = 3
    nx = 400
    ny = 400
    nz = 400
    xmin = 0
    xmax = 400
    ymin = 0
    ymax = 400
    zmin = 0
    zmax = 400
    elem_type = HEX8
  []
   parallel_type = distributed
[]
```
I am using 50 order parameters and 20K grains. In the past #17422, I did not face this issue. (I was using different HPC for this case.)

I tried following steps: 1) update Libmesh and MOOSE; 2) Install new MOOSE; and 3) Install new Moose with `./scripts/update_and_rebuild_petsc.sh` option (our HPC has installed PETSc for us). I am getting same error.

Any suggestions?

Thank you very much.

Kind regards

Vishal Yadav

**Comment by GiudGiud** [comment, marked as answer]
Hello

This should have worked..
This error should be linked to the petsc indexing scheme, and since you ran update_and_rebuild_petsc, this should have been using 64 bit indices. 

Could you please check that script, `update_and_rebuild_petsc` and see if the ` --with-64-bit-indices` is present in the ./configure call for petsc 

Guillaume

**Reply by ykvishal** [reply]
Thank  you @GiudGiud !

This suggestion worked.  Now, I can start this simulation.
